### PR TITLE
Instrumenting Cohere

### PR DESCRIPTION
**Pull request by HowieG**
Resolves ENG-365

### CODE REVIEW

**Comment by siyangqiu** [comment]
Can you check why the test are failing?

File: agentops/llm_tracker.py [hunk 1]
```diff
@@ -25,6 +26,11 @@ class LlmTracker:
                 "ChatCompletion.create",
                 "ChatCompletion.acreate",
             ),
+        },
+        'cohere': {
+            '1.0.0': (
```
Comment (areibman) [inline agentops/llm_tracker.py]: Shouldn't this be 5.0.0 for clarity sake? I know it's not at tight standard

File: agentops/llm_tracker.py [hunk 2]
```diff
@@ -95,13 +101,9 @@ def generator():
...
             return generator()
 
-        self.llm_event = LLMEvent(
-            init_timestamp=init_timestamp,
-            params=kwargs,
-            returns=response
-        )
         # v0.0.0 responses are dicts
         try:
+            self.llm_event.returns = response  # TODO: test
```
Comment (areibman) [inline agentops/llm_tracker.py]: test?

File: agentops/llm_tracker.py [hunk 3]
```diff
@@ -228,6 +226,116 @@ async def async_generator():
...
+
+                elif isinstance(chunk, StreamedChatResponse_TextGeneration):
+                    self.llm_event.completion += chunk.text
+
+            except Exception as e:
+                self.client.record(ErrorEvent(trigger_event=self.llm_event, exception=e))
+                kwargs_str = pprint.pformat(kwargs)
+                chunk = pprint.pformat(chunk)
+                logger.warning(
+                    f"🖇 AgentOps: Unable to parse a chunk for LLM call. Skipping upload to AgentOps\n"
```
Comment (areibman) [inline agentops/llm_tracker.py:286]: nit: saying agentops twice

File: agentops/llm_tracker.py [hunk 4]
```diff
@@ -350,6 +487,16 @@ def override_api(self):
...
 
+                if api == 'cohere':
+                    # Patch openai v1.0.0+ methods
+                    if hasattr(module, '__version__'):
+                        module_version = parse(module.__version__)
+                        if True:  # TODO: check version
+                            self.override_cohere_chat()
+                            self.override_cohere_chat_stream()
+                    else:
+                        logger.warning(f'🖇 AgentOps: Only Cohere>=x.x.x supported. v{module_version} found.')
```
Comment (areibman) [inline agentops/llm_tracker.py]: `x.x.x` -- provide a version number

File: tests/test_cohere.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import cohere
+import agentops
+from dotenv import load_dotenv
+load_dotenv()
+
+agentops.init(endpoint='http://localhost:8000')
```
Comment (areibman) [inline tests/test_cohere.py]: Remove localhost

File: tests/test_cohere.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+# agentops.init(endpoint='http://localhost:8000')
+# co = cohere.Client()
+
+# stream = co.chat_stream(
+#     message="Tell me a short story"
+# )
+
+# for event in stream:
+#     if event.event_type == "text-generation":
+#         print(event.text, end='')
```
Comment (areibman) [inline tests/test_cohere.py]: Remove

File: tests/test_cohere.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            "role": "CHATBOT",
+            "message": "The man who is widely credited with discovering gravity is Sir Isaac Newton",
+        },
+    ],
+    message="What year was he born?",
+    model="command"
+)
+
+print(chat)
```
Comment (areibman) [inline tests/test_cohere.py]: Show async/stream examples

File: README.md
Comment (areibman) [inline README.md:1]: Add cohere section to readme

File: agentops/llm_tracker.py [hunk 8]
```diff
@@ -228,6 +226,116 @@ async def async_generator():
 
         return response
 
+    def _handle_response_cohere(self, response, kwargs, init_timestamp):
+        # TODO: """Handle responses for Cohere versions >vx.x.x"""
```
Comment (areibman) [inline agentops/llm_tracker.py]: version x.x.x

**Review by areibman — DISMISSED** [review]

**Review by siyangqiu — APPROVED** [review]
Awesome work!

**Pull request merged**